### PR TITLE
use @Query and @Find for example code in chapter 4

### PR DESCRIPTION
**Pull request by gavinking**
instead of Query by Method Name, which has been exiled to a separate document.

### CODE REVIEW

File: spec/src/main/asciidoc/repository.asciidoc [hunk 1]
```diff
@@ -353,26 +355,32 @@ The following examples work through scenarios where static and dynamic sort crit
...
 ----
 // Sorts first by type. When type is the same, applies the Order's sort criteria
-Page<User> findByNameStartsWithOrderByType(String namePrefix,
-                                           PageRequest pagination,
-                                           Order<User> sorts);
+@Query("where left(name,length(?1)) = ?1")
+@OrderBy(_User.TYPE)
+Page<User> findByNamePrefix(String namePrefix,
+                            PageRequest pagination,
+                            Order<User> sorts);
```
Comment (njr-11) [inline spec/src/main/asciidoc/repository.asciidoc:362]: Prior to your changes, these examples covered 5 scenarios:

OrderBy keyword + Order param
OrderBy keyword + Sort... param
OrderBy anno + Order param
OrderBy anno + Sort... param
Query without ORDER BY + OrderBy anno + Order param

After changes, it now only covers 2 scenarios with several duplicated:

Query without ORDER BY + OrderBy anno + Order param
Query without ORDER BY + OrderBy anno + Sort... param
Query without ORDER BY + OrderBy anno + Order param
Query without ORDER BY + OrderBy anno + Sort... param
Query without ORDER BY + OrderBy anno + Order param

Adding suggestions to restore to previous:

```suggestion
Page<User> findByNameStartsWithOrderByType(String namePrefix,
                                           PageRequest pagination,
                                           Order<User> sorts);
```

File: spec/src/main/asciidoc/repository.asciidoc [hunk 2]
```diff
@@ -353,26 +355,32 @@ The following examples work through scenarios where static and dynamic sort crit
...
 
 // Sorts first by type. When type is the same, applies the criteria in the Sorts
-List<User> findByNameStartsWithOrderByType(String namePrefix, Sort<?>... sorts);
+@Query("where left(name,length(?1)) = ?1")
+@OrderBy(_User.TYPE)
+List<User> findByNamePrefix(String namePrefix, Sort<?>... sorts);
```
Comment (njr-11) [inline spec/src/main/asciidoc/repository.asciidoc:367]:
```suggestion
List<User> findByNameStartsWithOrderByType(String namePrefix, Sort<?>... sorts);
```

File: spec/src/main/asciidoc/repository.asciidoc [hunk 3]
```diff
@@ -353,26 +355,32 @@ The following examples work through scenarios where static and dynamic sort crit
...
 // Sorts first by age. When age is the same, applies the Order's sort criteria
-@OrderBy("age")
-Page<User> findByNameStartsWith(String namePrefix,
-                                PageRequest pagination,
-                                Order<User> sorts);
+@Query("where left(name,length(?1)) = ?1")
+@OrderBy(_User.AGE)
+Page<User> findByNamePrefix(String namePrefix,
+                            PageRequest pagination,
+                            Order<User> sorts);
```
Comment (njr-11) [inline spec/src/main/asciidoc/repository.asciidoc:374]:
```suggestion
@OrderBy(_User.AGE)
Page<User> findByNameStartsWith(String namePrefix,
                                PageRequest pagination,
                                Order<User> sorts);
```

File: spec/src/main/asciidoc/repository.asciidoc [hunk 4]
```diff
@@ -353,26 +355,32 @@ The following examples work through scenarios where static and dynamic sort crit
...
 
 // Sorts first by age. When age is the same, applies the criteria in the Sorts
-@OrderBy("age")
-List<User> findByNameStartsWith(String namePrefix, Sort<?>... sorts);
+@Query("where left(name,length(?1)) = ?1")
+@OrderBy(_User.AGE)
+List<User> findByNamePrefix(String namePrefix, Sort<?>... sorts);
```
Comment (njr-11) [inline spec/src/main/asciidoc/repository.asciidoc:379]:
```suggestion
@OrderBy(_User.AGE)
List<User> findByNameStartsWith(String namePrefix, Sort<?>... sorts);
```

**Review by njr-11 — CHANGES_REQUESTED** [review]

**Comment by gavinking** [comment]
@njr-11 Again, I don't think we should be having examples with QbMN in the main spec, because the main spec does not define what QbMN is. I agree that there is now duplication in the examples; the best fix to that would be to simply delete a couple of them.

**Comment by gavinking** [comment]
I mean, a reader of the Jakarta Data spec is going to encounter those examples and think "WTF is this `findByNameStartsWithOrderByType` stuff?"

**Comment by gavinking** [comment]
Please see 9c43087b0dfcac45f031a05758acfe10a26c01a8, which removes the duplication.

**Comment by njr-11** [comment]
> @njr-11 Again, I don't think we should be having examples with QbMN in the main spec, because the main spec does not define what QbMN is. I agree that there is now duplication in the examples; the best fix to that would be to simply delete a couple of them.

Given that Query by Method Name is still part of Jakarta Data, the spec has an obligation to cover how it behaves with respect to the rest of the specification, regardless of whether it is defined in a separate PDF or not. These examples are probably not that important and it could be stated that OrderBy keyword behaves the same as OrderBy annotation here, but I don't want to start a precedent for neglecting to adequately cover Query by Method Name because of where it appears in the spec.

**Comment by gavinking** [comment]
@njr-11 Then move the examples to the Query by Method Name document. I'm not sure they're needed, but I don't object to having them. I've put quite a lot of work into improving the material in that document over the weekend, but any help would be appreciated, which includes anything you think is missing and want to add.

But for sure these examples don't belong where they are right now, because the reader has never encountered a Query by Method Name, and will have no idea how to interpret the examples. Examples have to be examples of something you have properly introduced previously. You have to assume that the reader is reading the text in a linear fashion, even if that's not always the case in practice.

**Comment by gavinking** [comment]
I think this should be merged. 

I insist that we should never have examples in the spec which the reader will not understand. What is the point of such an example?

**Review by njr-11 — APPROVED** [review]

**Pull request merged**